explorer.go

```diff
@@ -138,20 +169,60 @@ func (s APIEndpoint) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		Timeout: skycoinRequestTimeout,
 	}
 
-	resp, err := c.Get(skycoinURL)
+    resp, err := c.Get(skycoinURL)
+    var cachedResponse []byte
+    var cachedResponseFound bool
 	if err != nil {
-		msg := "Request to skycoin node failed"
-		log.Println("ERROR:", msg, skycoinURL)
-		http.Error(w, msg, http.StatusInternalServerError)
-		return
+        // Cancel the operation, but not if a previous cached value is available
+        cachedData, found := cacheMgr.Get(s.ExplorerPath)
+        cachedResponseFound = found;
+
+        if cachedResponseFound {
+            cachedResponse = cachedData.([]byte)
+        } else {
+            msg := "Request to skycoin node failed"
+            log.Println("ERROR:", msg, skycoinURL)
+            http.Error(w, msg, http.StatusInternalServerError)
+            return
+        }
 	}
 
-	defer resp.Body.Close()
+    var responseBody io.Reader
 
-	w.WriteHeader(resp.StatusCode)
-	w.Header().Set("Content-Type", "application/json")
+    if err != nil {
```

Comment:
It should always try to make the request and use the cache as a fallback

Reply:
The code already worked that way. However, with the changes made with fmt the code is a bit confusing to read on Github, so I added 3 comments below to explain a little bit better how the process works.